main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -2074,7 +2077,7 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
       PredefinedClasses.lookupClassSym("Hash", root),
       PredefinedClasses.lookupClassSym("ToString", root),
     )
-    classes.map(Ast.TypeConstraint(_, tpe))
+    classes.map(Ast.TypeConstraint(_, tpe, SourceLocation.Unknown))
```

Comment:
I think here and `mkTypeClassConstraintsForLatticeTerm` we should take a loc parameter. Otherwise it will be hard to understand as a programmer.